metricbeat/module/kafka/broker.go

```diff
@@ -134,6 +137,13 @@ func (b *Broker) Connect() error {
 	debugf("found matching broker %v with id %v", other.Addr(), other.ID())
 	b.id = other.ID()
 	b.advertisedAddr = other.Addr()
+
+	c, err := getClusteWideClient(b.Addr(), b.cfg)
```

Comment:
I was thinking that we could use this client for everything, but not, we may still need to fetch offsets from non-leader partition replicas for monitoring purpouses.

Reply:
Though we can also use `client.Leader(topic, partitionID)` for that.

Reply:
+1 in revisiting this in a followup PR with refactoring purpose